smartlearner/optimizers/adam.py

```diff
@@ -0,0 +1,43 @@
+from collections import OrderedDict
+
+import theano
+import theano.tensor as T
+import numpy as np
+
+from ..interfaces import Optimizer
+from ..utils import sharedX
+
+class Adam(Optimizer):
+    def __init__(self, loss, learning_rate=0.0001, b1=0.9, b2=0.999, epsilon=1e-8, lmbda=(1 - 1e-8)):
```

Comment:
Also, do you know what `b1`, `b2` and the `epsilon` do? Should add short docstring for them (I know we should do it for `AdaGrad` too :P).